// @mentra/sdk/dist/constants/log-messages/updates.js
"use strict";
/**
 * updates.ts
 *
 * This file defines constant messages that should be displayed
 * in the terminal to notify developers about new SDK releases.
 *
 * Each function generates a stylized ASCII message (banner-style)
 * that highlights the latest SDK version and provides the npm install command.
 * https://patorjk.com/software/taag/
 *
 * These messages are intended to be logged to the console or shown in
 * terminal output so developers are aware of updates in a clear
 * and visually distinct way.
 *
 *
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.newSDKUpdate = void 0;
const chalk_1 = __importDefault(require("chalk"));
const boxen_1 = __importDefault(require("boxen"));
const logos_1 = require("./logos");
const createUpdateNotification = (versionNumb) => {
    const line = chalk_1.default.bold.gray("-------------------------------");
    const logo = chalk_1.default.cyan(logos_1.mentraLogo_1);
    const title = chalk_1.default.bold.cyan(logos_1.newUpdateText);
    const versionMessage = `Version ${chalk_1.default.bold.cyan(`SDK VERSION V${versionNumb} is out! 🎉`)}`;
    const currentNote = chalk_1.default.yellow("You are running an older version");
    const instructions = chalk_1.default.yellow("Update to the latest version with:");
    const command = chalk_1.default.green.bold("bun install @mentra/sdk@latest");
    const content = [
        logo,
        title,
        line,
        versionMessage,
        currentNote,
        line,
        instructions,
        command,
    ].join("\n");
    return (0, boxen_1.default)(content, {
        padding: 1,
        margin: 1,
        borderStyle: "round",
        borderColor: "cyan",
        textAlignment: "left",
    });
};
const newSDKUpdate = (versionNumb) => {
    return createUpdateNotification(versionNumb);
};
exports.newSDKUpdate = newSDKUpdate;
