// @mentra/sdk/dist/constants/log-messages/warning.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.cameraWarn = exports.postNotficationWarn = exports.readNotficationWarn = exports.calendarWarn = exports.baackgroundLocationWarn = exports.locationWarn = exports.noMicrophoneWarn = void 0;
/**
 * warning.ts
 *
 * This file defines styled warning messages that are displayed when an app
 * attempts to use functionality requiring permissions it hasn't declared.
 *
 * Each function generates a bordered terminal warning box with:
 * - ASCII art logo (left side)
 * - Permission requirement details (right side)
 * - Link to developer portal for adding permissions
 *
 * The warnings use chalk for terminal colors and boxen for bordered output,
 * creating a professional side-by-side layout that alerts developers to
 * missing permissions in their app configuration.
 *
 * These are shown during SDK runtime when permission checks fail, helping
 * developers identify and fix permission issues quickly.
 */
const chalk_1 = __importDefault(require("chalk"));
const boxen_1 = __importDefault(require("boxen"));
const logos_1 = require("./logos");
const createPermissionWarning = (permissionName, funcName, packageName) => {
    // Strip ANSI codes for width calculation
    // eslint-disable-next-line no-control-regex
    const stripAnsi = (str) => str.replace(/\u001b\[\d+m/g, "");
    const title = chalk_1.default.bold.yellow("⚠️  Permission Required");
    const message = `${chalk_1.default.yellow(funcName || "This function")} requires ${chalk_1.default.bold(permissionName)} permission.`;
    const instructions = chalk_1.default.dim("Please enable this permission in the developer portal at:");
    const url = chalk_1.default.cyan.underline(`https://console.mentra.glass/apps/${packageName}/edit`);
    const hint = chalk_1.default.dim("under *Required Permissions*.");
    // Split logo into lines (without color for width calculation)
    const logoLines = logos_1.warnLog.split("\n");
    const coloredLogoLines = logoLines.map((line) => chalk_1.default.yellow(line));
    const textContent = [title, "", message, "", instructions, url, "", hint];
    // Find actual logo width (max line length without ANSI codes)
    const logoWidth = Math.max(...logoLines.map((line) => stripAnsi(line).length));
    // Create side-by-side layout
    const maxLines = Math.max(coloredLogoLines.length, textContent.length);
    const combinedLines = [];
    for (let i = 0; i < maxLines; i++) {
        const rawLogoLine = logoLines[i] || "";
        const coloredLogoLine = coloredLogoLines[i] || "";
        const actualLogoLength = stripAnsi(rawLogoLine).length;
        const padding = " ".repeat(Math.max(0, logoWidth - actualLogoLength));
        const textLine = textContent[i] || "";
        combinedLines.push(`${coloredLogoLine}${padding}    ${textLine}`);
    }
    return (0, boxen_1.default)(combinedLines.join("\n"), {
        padding: 1,
        margin: 1,
        borderStyle: "round",
        borderColor: "yellow",
        float: "left",
    });
};
const noMicrophoneWarn = (funcName, packageName) => {
    return createPermissionWarning("microphone", funcName, packageName);
};
exports.noMicrophoneWarn = noMicrophoneWarn;
const locationWarn = (funcName, packageName) => {
    return createPermissionWarning("location", funcName, packageName);
};
exports.locationWarn = locationWarn;
const baackgroundLocationWarn = (funcName, packageName) => {
    return createPermissionWarning("background location", funcName, packageName);
};
exports.baackgroundLocationWarn = baackgroundLocationWarn;
const calendarWarn = (funcName, packageName) => {
    return createPermissionWarning("calendar", funcName, packageName);
};
exports.calendarWarn = calendarWarn;
const readNotficationWarn = (funcName, packageName) => {
    return createPermissionWarning("read notification", funcName, packageName);
};
exports.readNotficationWarn = readNotficationWarn;
const postNotficationWarn = (funcName, packageName) => {
    return createPermissionWarning("post notification", funcName, packageName);
};
exports.postNotficationWarn = postNotficationWarn;
const cameraWarn = (funcName, packageName) => {
    return createPermissionWarning("camera", funcName, packageName);
};
exports.cameraWarn = cameraWarn;
