// @mentra/sdk/dist/utils/permissions-utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.cameraWarnLog = exports.postNotificationWarnLog = exports.readNotificationWarnLog = exports.calendarWarnLog = exports.backgroundLocationWarnLog = exports.locationWarnLog = exports.microPhoneWarnLog = void 0;
/**
 * permissions-utils.ts
 *
 * This file provides runtime permission validation utilities for the MentraOS SDK.
 *
 * Each function queries the public permissions API endpoint to check if an app
 * has declared the required permission for a specific feature. If the permission
 * is missing, a styled warning message is displayed in the terminal.
 *
 * Key features:
 * - Fetches app permissions from /api/public/permissions/:packageName
 * - Gracefully handles offline/unreachable endpoints (silent failure)
 * - Displays professional bordered warnings when permissions are missing
 * - Non-blocking - allows app execution to continue even if checks fail
 *
 * These functions are called automatically by SDK methods that require specific
 * permissions (e.g., microphone access, location tracking, camera, etc.) to help
 * developers identify missing permission declarations during development.
 */
const warning_1 = require("../constants/log-messages/warning");
// Check if app has microphone permission, warn if missing
const microPhoneWarnLog = (cloudServerUrl, packageName, funcName) => {
    if (!cloudServerUrl)
        return;
    const permissionsUrl = `${cloudServerUrl}/api/public/permissions/${encodeURIComponent(packageName)}`;
    // console.log(`Fetching permissions from: ${permissionsUrl}`);
    fetch(permissionsUrl)
        .then(async (res) => {
        const contentType = res.headers.get("content-type");
        if (!res.ok) {
            console.warn(`Permission API returned ${res.status}: ${res.statusText}`);
            return null;
        }
        if (contentType && contentType.includes("application/json")) {
            return (await res.json());
        }
        else {
            const text = await res.text();
            console.warn(`Permission API returned non-JSON response: ${text}`);
            return null;
        }
    })
        .then((data) => {
        if (data) {
            const hasMic = data.permissions.some((p) => p.type === "MICROPHONE");
            if (!hasMic) {
                console.log((0, warning_1.noMicrophoneWarn)(funcName, packageName));
            }
        }
    })
        .catch((err) => {
        // Silently fail if endpoint is unreachable - don't block execution
        console.debug("Permission check skipped - endpoint unreachable:", err.message);
    });
};
exports.microPhoneWarnLog = microPhoneWarnLog;
// Check if app has location permission, warn if missing
const locationWarnLog = (cloudServerUrl, packageName, funcName) => {
    if (!cloudServerUrl)
        return;
    const permissionsUrl = `${cloudServerUrl}/api/public/permissions/${encodeURIComponent(packageName)}`;
    fetch(permissionsUrl)
        .then(async (res) => {
        const contentType = res.headers.get("content-type");
        if (!res.ok) {
            console.warn(`Permission API returned ${res.status}: ${res.statusText}`);
            return null;
        }
        if (contentType && contentType.includes("application/json")) {
            return (await res.json());
        }
        else {
            const text = await res.text();
            console.warn(`Permission API returned non-JSON response: ${text}`);
            return null;
        }
    })
        .then((data) => {
        if (data) {
            const hasLocation = data.permissions.some((p) => p.type === "LOCATION");
            if (!hasLocation) {
                console.log((0, warning_1.locationWarn)(funcName, packageName));
            }
        }
    })
        .catch((err) => {
        console.debug("Permission check skipped - endpoint unreachable:", err.message);
    });
};
exports.locationWarnLog = locationWarnLog;
// Check if app has background location permission, warn if missing
const backgroundLocationWarnLog = (cloudServerUrl, packageName, funcName) => {
    if (!cloudServerUrl)
        return;
    const permissionsUrl = `${cloudServerUrl}/api/public/permissions/${encodeURIComponent(packageName)}`;
    fetch(permissionsUrl)
        .then(async (res) => {
        const contentType = res.headers.get("content-type");
        if (!res.ok) {
            console.warn(`Permission API returned ${res.status}: ${res.statusText}`);
            return null;
        }
        if (contentType && contentType.includes("application/json")) {
            return (await res.json());
        }
        else {
            const text = await res.text();
            console.warn(`Permission API returned non-JSON response: ${text}`);
            return null;
        }
    })
        .then((data) => {
        if (data) {
            const hasBackgroundLocation = data.permissions.some((p) => p.type === "BACKGROUND_LOCATION");
            if (!hasBackgroundLocation) {
                console.log((0, warning_1.baackgroundLocationWarn)(funcName, packageName));
            }
        }
    })
        .catch((err) => {
        console.debug("Permission check skipped - endpoint unreachable:", err.message);
    });
};
exports.backgroundLocationWarnLog = backgroundLocationWarnLog;
// Check if app has calendar permission, warn if missing
const calendarWarnLog = (cloudServerUrl, packageName, funcName) => {
    if (!cloudServerUrl)
        return;
    const permissionsUrl = `${cloudServerUrl}/api/public/permissions/${encodeURIComponent(packageName)}`;
    fetch(permissionsUrl)
        .then(async (res) => {
        const contentType = res.headers.get("content-type");
        if (!res.ok) {
            console.warn(`Permission API returned ${res.status}: ${res.statusText}`);
            return null;
        }
        if (contentType && contentType.includes("application/json")) {
            return (await res.json());
        }
        else {
            const text = await res.text();
            console.warn(`Permission API returned non-JSON response: ${text}`);
            return null;
        }
    })
        .then((data) => {
        if (data) {
            const hasCalendar = data.permissions.some((p) => p.type === "CALENDAR");
            if (!hasCalendar) {
                console.log((0, warning_1.calendarWarn)(funcName, packageName));
            }
        }
    })
        .catch((err) => {
        console.debug("Permission check skipped - endpoint unreachable:", err.message);
    });
};
exports.calendarWarnLog = calendarWarnLog;
// Check if app has read notifications permission, warn if missing
const readNotificationWarnLog = (cloudServerUrl, packageName, funcName) => {
    if (!cloudServerUrl)
        return;
    const permissionsUrl = `${cloudServerUrl}/api/public/permissions/${encodeURIComponent(packageName)}`;
    fetch(permissionsUrl)
        .then(async (res) => {
        const contentType = res.headers.get("content-type");
        if (!res.ok) {
            console.warn(`Permission API returned ${res.status}: ${res.statusText}`);
            return null;
        }
        if (contentType && contentType.includes("application/json")) {
            return (await res.json());
        }
        else {
            const text = await res.text();
            console.warn(`Permission API returned non-JSON response: ${text}`);
            return null;
        }
    })
        .then((data) => {
        if (data) {
            const hasReadNotifications = data.permissions.some((p) => p.type === "READ_NOTIFICATIONS");
            if (!hasReadNotifications) {
                console.log((0, warning_1.readNotficationWarn)(funcName, packageName));
            }
        }
    })
        .catch((err) => {
        console.debug("Permission check skipped - endpoint unreachable:", err.message);
    });
};
exports.readNotificationWarnLog = readNotificationWarnLog;
// Check if app has post notifications permission, warn if missing
const postNotificationWarnLog = (cloudServerUrl, packageName, funcName) => {
    if (!cloudServerUrl)
        return;
    const permissionsUrl = `${cloudServerUrl}/api/public/permissions/${encodeURIComponent(packageName)}`;
    fetch(permissionsUrl)
        .then(async (res) => {
        const contentType = res.headers.get("content-type");
        if (!res.ok) {
            console.warn(`Permission API returned ${res.status}: ${res.statusText}`);
            return null;
        }
        if (contentType && contentType.includes("application/json")) {
            return (await res.json());
        }
        else {
            const text = await res.text();
            console.warn(`Permission API returned non-JSON response: ${text}`);
            return null;
        }
    })
        .then((data) => {
        if (data) {
            const hasPostNotifications = data.permissions.some((p) => p.type === "POST_NOTIFICATIONS");
            if (!hasPostNotifications) {
                console.log((0, warning_1.postNotficationWarn)(funcName, packageName));
            }
        }
    })
        .catch((err) => {
        console.debug("Permission check skipped - endpoint unreachable:", err.message);
    });
};
exports.postNotificationWarnLog = postNotificationWarnLog;
// Check if app has camera permission, warn if missing
const cameraWarnLog = (cloudServerUrl, packageName, funcName) => {
    if (!cloudServerUrl)
        return;
    const permissionsUrl = `${cloudServerUrl}/api/public/permissions/${encodeURIComponent(packageName)}`;
    fetch(permissionsUrl)
        .then(async (res) => {
        const contentType = res.headers.get("content-type");
        if (!res.ok) {
            console.warn(`Permission API returned ${res.status}: ${res.statusText}`);
            return null;
        }
        if (contentType && contentType.includes("application/json")) {
            return (await res.json());
        }
        else {
            const text = await res.text();
            console.warn(`Permission API returned non-JSON response: ${text}`);
            return null;
        }
    })
        .then((data) => {
        if (data) {
            const hasCamera = data.permissions.some((p) => p.type === "CAMERA");
            if (!hasCamera) {
                console.log((0, warning_1.cameraWarn)(funcName, packageName));
            }
        }
    })
        .catch((err) => {
        console.debug("Permission check skipped - endpoint unreachable:", err.message);
    });
};
exports.cameraWarnLog = cameraWarnLog;
